internal/graph/check_test.go

```diff
@@ -2998,3 +3005,531 @@ func TestIteratorReadStartingFromUser(t *testing.T) {
+					relations
+						define member: [user]
+				type document
+					relations
+						define viewer: [group#member]`),
+			tuples: []dsResults{
+				{
+					tuples: []*openfgav1.TupleKey{},
+					err:    fmt.Errorf("mock_error"),
+				},
+				{
+					tuples: []*openfgav1.TupleKey{
+						tuple.NewTupleKey("group:11", "member", "user:maria"),
+						tuple.NewTupleKey("group:12", "member", "user:maria"),
+						tuple.NewTupleKey("group:13", "member", "user:maria"),
+						tuple.NewTupleKey("group:14", "member", "user:maria"),
+					},
+					err: nil,
+				},
+			},
+			usersetsChannelResult: []usersetsChannelStruct{
+				{
+					err:            nil,
+					objectRelation: "group#member",
+					objectIDs:      []string{"1", "2", "3"},
+				},
+				{
+					err:            nil,
+					objectRelation: "group#member",
+					objectIDs:      []string{"10", "11", "12"},
+				},
+			},
+			ctxCancelled: false,
+			expectedResolveCheckResponse: &ResolveCheckResponse{
+				Allowed: true,
+				ResolutionMetadata: &ResolveCheckResponseMetadata{
+					DatastoreQueryCount: 2,
+				},
+			},
+			errorExpected: nil,
+		},
+		{
+			name: "iterator_error_first_batch_but_not_found_second",
+			model: parser.MustTransformDSLToProto(`
+				model
+					schema 1.1
+				type user
+				type group
+					relations
+						define member: [user]
+				type document
+					relations
+						define viewer: [group#member]`),
+			tuples: []dsResults{
+				{
+					tuples: []*openfgav1.TupleKey{},
+					err:    fmt.Errorf("mock_error"),
+				},
+				{
+					tuples: []*openfgav1.TupleKey{
+						tuple.NewTupleKey("group:11", "member", "user:maria"),
+						tuple.NewTupleKey("group:12", "member", "user:maria"),
+						tuple.NewTupleKey("group:13", "member", "user:maria"),
+						tuple.NewTupleKey("group:14", "member", "user:maria"),
+					},
+					err: nil,
+				},
+			},
+			usersetsChannelResult: []usersetsChannelStruct{
+				{
+					err:            nil,
+					objectRelation: "group#member",
+					objectIDs:      []string{"1", "2", "3"},
+				},
+				{
+					err:            nil,
+					objectRelation: "group#member",
+					objectIDs:      []string{"4", "4", "6"},
+				},
+			},
+			ctxCancelled:                 false,
+			expectedResolveCheckResponse: nil,
+			errorExpected:                fmt.Errorf("mock_error"),
+		},
+		{
+			name: "userset_chan_error",
+			model: parser.MustTransformDSLToProto(`
+				model
+					schema 1.1
+				type user
+				type group
+					relations
+						define member: [user]
+				type document
+					relations
+						define viewer: [group#member]`),
+			tuples: []dsResults{},
+			usersetsChannelResult: []usersetsChannelStruct{
+				{
+					err:            fmt.Errorf("mock_error"),
+					objectRelation: "group#member",
+					objectIDs:      []string{"0", "2", "8"},
+				},
+			},
+			ctxCancelled:                 false,
+			expectedResolveCheckResponse: nil,
+			errorExpected:                fmt.Errorf("mock_error"),
+		},
+		{
+			name: "ttu_tuple_found",
```

Comment:
Why did we add a case specifically for TTU? There is nothing in the code under test (`consumeUsersets`) that deals with TTU, and nothing in `buildCheckAssociatedObjects` either

If you decide to remove this test please put the model in one variable so that we don't have to declare it in every test

Reply:
nuked